src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs

```diff
@@ -106,51 +108,65 @@ protected async Task<bool> SyncBatchOfBlocksAsync(CancellationToken cancellation
             // API method that provides blocks should't give us blocks that are not mature!
             var model = new MaturedBlockRequestModel(this.store.NextMatureDepositHeight, blocksToRequest);
 
-            this.logger.LogDebug("Request model created: {0}:{1}, {2}:{3}.", nameof(model.BlockHeight), model.BlockHeight,
-                nameof(model.MaxBlocksToSend), model.MaxBlocksToSend);
+            this.logger.LogDebug("Request model created: {0}:{1}, {2}:{3}.", nameof(model.BlockHeight), model.BlockHeight, nameof(model.MaxBlocksToSend), model.MaxBlocksToSend);
 
             // Ask for blocks.
-            IList<MaturedBlockDepositsModel> matureBlockDeposits = await this.federationGatewayClient.GetMaturedBlockDepositsAsync(model, cancellationToken).ConfigureAwait(false);
+            ApiResult<List<MaturedBlockDepositsModel>> matureBlockDepositsResult = await this.federationGatewayClient.GetMaturedBlockDepositsAsync(model, cancellationToken).ConfigureAwait(false);
 
-            bool delayRequired = true;
+            if (matureBlockDepositsResult == null)
+            {
+                this.logger.LogInformation("Unable to fetch mature block deposits from the counter chain node; {0} didn't respond.", this.federationGatewayClient.EndpointUrl);
+                this.logger.LogTrace("(-)[COUNTER_CHAIN_NODE_UNRESPONSIVE]:true");
+                return true;
+            }
 
-            if (matureBlockDeposits != null)
+            if (matureBlockDepositsResult.Value == null)
             {
-                // Log what we've received.
-                foreach (MaturedBlockDepositsModel maturedBlockDeposit in matureBlockDeposits)
-                {
-                    // Order transactions in block deterministically
-                    maturedBlockDeposit.Deposits = maturedBlockDeposit.Deposits.OrderBy(x => x.Id,
-                        Comparer<uint256>.Create(DeterministicCoinOrdering.CompareUint256)).ToList();
+                this.logger.LogInformation("Unable to fetch mature block deposits from the counter chain node; mature deposits was null.");
+                this.logger.LogTrace("(-)[COUNTER_CHAIN_NODE_RESULT_NULL]:true");
+                return true;
+            }
 
-                    foreach (IDeposit deposit in maturedBlockDeposit.Deposits)
-                    {
-                        this.logger.LogDebug("New deposit received BlockNumber={0}, TargetAddress='{1}', depositId='{2}', Amount='{3}'.",
-                            deposit.BlockNumber, deposit.TargetAddress, deposit.Id, deposit.Amount);
-                    }
-                }
+            if (!matureBlockDepositsResult.Succeeded)
```

Comment:
I think this actually occurs quite a lot but it's expected. This could be annoying. Let us know what you find in your tests but my guess is virtually every 10 seconds we'll get this message for `height xxxx is not mature enough yet`